civicrm.install

```diff
@@ -22,13 +24,25 @@ function civicrm_install() {
 
   $installed = $setup->checkInstalled();
   if ($installed->isSettingInstalled() || $installed->isDatabaseInstalled()) {
-    drupal_set_message(t("CiviCRM appears to have already been installed. Skipping full installation."));
+    \Drupal::messenger()
+      ->addMessage(t("CiviCRM appears to have already been installed. Skipping full installation."));
```

Comment:
I still think this needs to throw an exception not a message, because in the situation you end up here you would never see this message because drupal will continue on and run into a different problem that throws a different exception, so it's harder to track down that it actually failed here.

Reply:
Warning is sufficient?

Reply:
But when you try actually running it in the situation where this comes up, the message is never displayed because a later exception ends up getting displayed instead.

Reply:
I agree, this needs to throw an exception.

Reply:
@MikeyMJCO see also https://github.com/civicrm/civicrm-drupal-8/pull/53/files

Reply:
Based on that I think we need to merge #52 and #53 together!